libtcc.c

```diff
@@ -834,6 +834,10 @@ LIBTCCAPI TCCState *tcc_new(void)
     tcc_define_symbol(s, "__riscv_float_abi_double", NULL);
 #endif
 
+#if defined(TCC_TARGET_ARM64) && defined(HAVE_PTRAUTH)
+    tcc_define_symbol(s, "HAVE_PTRAUTH", NULL);
+#endif
+
```

Comment:
Would be good to move this to the `TCC_TARGET_ARM64` region above, and name the define something like `__aarch64_ptrauth__`, as all built-in defines (at this level) should be prefixed with underscores.